### PR TITLE
Ruler: add configuration option to disable event-based sync

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

The event-based rule groups re-sync introduced in #4975 is expected to benefit Mimir clusters with a large number of tenants, not updating rule groups frequently. However, there are some cases where a single-tenant Mimir cluster may get rule groups re-uploaded very frequently and so the event-based re-sync may actually increase the cost instead of decreasing it.

In this PR I'm simply introducing an answer to disable it, while keeping it enabled by default. To do it, I've took the simplest path: do not put the request to the outbound queue when the event-based sync is disabled. The queue and queue processor are started anyway, but they're not expected to make any dent in terms of resources utilization given they're not going to be used.

#### Which issue(s) this PR fixes or relates to

Part of #4941.

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pracucci** [comment]
> Would it make sense to have this flag per tenant, so that we can avoid syncing rules when a tenant performs API change and only do it via regular (time-based) sync?

Makes sense. Done in https://github.com/grafana/mimir/pull/5053/commits/ca53d3118cc53a3666dd37e935921f620fa28f5a.

**Review by gotjosh — APPROVED** [review]
LGTM, I was going to suggest the same as peter - if we're initializing all the components anyways might as well just make it a runtime configuration via limits. Great job 🙏

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,7 +46,7 @@
...
 * [ENHANCEMENT] Querier: only use the minimum set of chunks from ingesters when querying, and cancel unnecessary requests to ingesters sooner if we know their results won't be used. #5016
 * [ENHANCEMENT] Add `-enable-go-runtime-metrics` flag to expose all go runtime metrics as Prometheus metrics. #5009
-* [ENHANCEMENT] Ruler: trigger a synchronization of tenant's rule groups as soon as they change the rules configuration via API. This synchronization is in addition of the periodic syncing done every `-ruler.poll-interval`. #4975
+* [ENHANCEMENT] Ruler: trigger a synchronization of tenant's rule groups as soon as they change the rules configuration via API. This synchronization is in addition of the periodic syncing done every `-ruler.poll-interval`. The new behavior is enabled by default, but can be disabled with `-ruler.sync-rules-on-changes-enabled=false` (configurable on a per-tenant basis too). #4975 #5053
```
[the remark this answers is not in the document]
Comment (pracucci) [reply CHANGELOG.md:49]: The flag is defined in limits. Every "limit" (runtime config) can both be set globally via CLI flag and YAML, and also on a per-tenant basis via overrides (YAML only).

[the remark this answers is not in the document]
Comment (gotjosh) [reply CHANGELOG.md:49]: Ah, that makes sense TIL.

**Pull request merged**